cmd/otel-allocator/watcher/promOperator.go

```diff
@@ -76,43 +96,78 @@ func NewPrometheusCRWatcher(logger logr.Logger, cfg allocatorconfig.Config) (*Pr
 		return nil, err
 	}
 
-	servMonSelector := getSelector(cfg.ServiceMonitorSelector)
+	store := assets.NewStore(clientset.CoreV1(), clientset.CoreV1())
+	promRegisterer := prometheusgoclient.NewRegistry()
+	operatorMetrics := operator.NewMetrics(promRegisterer)
 
-	podMonSelector := getSelector(cfg.PodMonitorSelector)
+	nsMonInf, err := getNamespaceInformer(ctx, map[string]struct{}{v1.NamespaceAll: {}}, promOperatorLogger, clientset, operatorMetrics)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
It looks like we error if we don't have a permission to view namespaces. Could we instead log a warning here? Otherwise this will be a breaking change that will require all Target Allocator users to update their roles. I'd rather avoid that. This is how the allocator has behaved thus far as well - if it didn't have permission to view some resource, it'd log a warning and continue.

Reply:
in said note we should be explicit on the changes the user needs to make i.e. role -> clusterrole & add namespace.

Reply:
Thanks @swiatekm-sumo and @jaronoff97.
I did update the PR to handle the case where there are no namespace list/watch permissions. Please let me know if this looks good. With this change, the TA will continue to run but will not load any config since the resourceselector doesnt get initialized.

Reply:
Following up. @swiatekm-sumo @jaronoff97 - Pls let me know if you would like any other changes. Thanks!

Reply:
@swiatekm-sumo - Sorry I didnt see this in time as I am in PST. But I agree that we should ensure we don't break customers with this change.